### PR TITLE
[react-native-reanimated] Upgrade to alpha.9

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

SDK 40 is coming to town!

# How

1. `et uvm -m react-native-reanimated`
2. Changed some iOS imports so they match how files are structured (instead of using a library/subproject/pod we put all the files in one directory).
3. Changed the way JS executor is instantiated so it sets the reanimated proxy under `__reanimatedModuleProxy` (used @Szymon20000's https://github.com/Szymon20000/expo/commit/01cb5aa9cc9e1b34f9b5471abb4e937b9547e918 and `UIResponder+Reanimated.mm` as guidance).

# Test Plan

Ran Expo client for both iOS and Android, both compiled, ran Reanimated v2 demo in a separate project with JS code updated to `2.0.0-alpha.9`, both animated width of the block nicely.

### CODE REVIEW

**Review by bbarthec — APPROVED** [review]
Ok, looks ok to me, we'll see how it runs during the QA and #dogfooding 😉

**Review by bbarthec — COMMENTED** [review]
I believe we need some CHANGELOG entry, but I'm not sure how we communicate `reanimated v2` support 🤔

**Comment by sjchmiela** [comment]
Actually I did a terrible job last SDK when adding this feature to Expo SDK not taking care of any advertisement steps, so apart from adding a note to blog post and docs (h/t @brentvatne for taking care of that) there is no historical reference to adding Reanimated v2 in CHANGELOGs.

That being said I think it should be included in root changelog so that's where I'm adding it now.

File: docs/pages/versions/unversioned/sdk/reanimated.md [hunk 1]
```diff
@@ -22,23 +22,11 @@ The second major version of this library offers a much easier API, along with si
 
 > ⏩ If you want to play with v2-alpha in a new project before adding it to an existing project, run `npx crna --template with-reanimated2` to create a project with it configured and ready to use.
 
-To opt in to using TurboModules for react-native-reanimated, add the `experiments.turboModules` key to your `app.json`:
```
Comment (brentvatne) [inline docs/pages/versions/unversioned/sdk/reanimated.md:25]: is it no longer needed to enable turboModules explicitly with this flag?

Comment (sjchmiela) [reply docs/pages/versions/unversioned/sdk/reanimated.md:25]: That is correct, the reanimated team has changed the way their native code is integrated and as they exposed a property on `global` on Android in SDK39, they do so now on iOS too.

See first note at https://docs.swmansion.com/react-native-reanimated/docs/installation#ios

Comment (brentvatne) [reply docs/pages/versions/unversioned/sdk/reanimated.md:25]: amazing! thanks for the info

**Review by brentvatne — APPROVED** [review]

**Pull request merged**